extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseQuickpick.ts

```diff
@@ -32,19 +32,26 @@ export async function createNewProjectFromDatabaseWithQuickpick(connectionInfo?:
 	let dbs: string[] | undefined = undefined;
 	while (!dbs) {
 		// Get the list of databases now to validate that the connection is valid and re-prompt them if it isn't
-		try {
-			connectionUri = await vscodeMssqlApi.connect(connectionProfile);
-			dbs = (await vscodeMssqlApi.listDatabases(connectionUri))
-				.filter(db => !constants.systemDbs.includes(db)); // Filter out system dbs
-		} catch (err) {
-			// The mssql extension handles showing the error to the user. Prompt the user
-			// for a new connection and then go and try getting the DBs again
+		let shouldPromptForConnection = connectionProfile?.server === undefined;
```

Comment:
please add a comment for the case when `connectionProfile.server` is undefined

Reply:
Try setting this outside while loop to avoid resetting in next round - so you can support 'else' case.

Reply:
I _think_ it's correct without the `else` case (see [my reply to the other thread](https://github.com/microsoft/azuredatastudio/pull/23159/files#r1197087131)), but it's a good call to move this outside the loop nonetheless.